prae/src/core.rs

```diff
@@ -2,22 +2,51 @@ use core::hash::Hash;
 use std::fmt;
 use std::ops::{Deref, Index};
 
-/// Default validation error. It is used for [`define!`](prae_macro::define) macro with `ensure`
-/// keyword.
-#[derive(PartialEq)]
-pub struct ValidationError;
+/// Used for [`define!`](prae_macro::define) macro with `ensure` keyword.
+#[derive(Clone)]
+pub struct ConstructionError<T> 
+where
+    T: fmt::Debug
+{
+    /// The name of the type where this ConstructionError originated.
+    guarded_type_name: String,
```

Comment:
Not sure it should be `String` since we know the type during compile time

Reply:
I want to print the alias, how would I store that?

Reply:
The type is quite a mouthful :smile:

Reply:
but.. it could be a `&'static str`...

Reply:
Nice catch, the change is pushed.

Reply:
Yeah, I meant the `&'static str`, sorry for not being clear